pkg/realizer/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 VMware
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package healthcheck
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/vmware-tanzu/cartographer/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/cartographer/pkg/conditions"
+	"github.com/vmware-tanzu/cartographer/pkg/utils"
+)
+
+func DetermineHealthCondition(rule *v1alpha1.HealthRule, realizedResource *v1alpha1.RealizedResource, stampedObject *unstructured.Unstructured) metav1.Condition {
+	if rule == nil {
+		if realizedResource == nil {
+			return conditions.NoResourceResourcesHealthyCondition()
+		} else if len(realizedResource.Outputs) > 0 {
+			return conditions.OutputAvailableResourcesHealthyCondition()
+		} else if realizedResource.TemplateRef != nil && realizedResource.TemplateRef.Kind == "ClusterTemplate" && realizedResource.TemplateRef.APIVersion == "carto.run/v1alpha1" {
```

Comment:
uuuuhh it seems a little fragile to have these strings right here 😬 

wdyt of leveraging v1alpha1.SchemaGroupVersion

https://github.com/vmware-tanzu/cartographer/blob/693036beb08f237ad45e3715c92a61a7e718cd81/pkg/apis/v1alpha1/groupversion_info.go#L23-L26

for APIVersion?

given that various `&&`,  might be useful to turn the predicate into a small function that returns a bool, wdyt?